strawberryfields/circuitspecs/circuit_specs.py

```diff
@@ -380,3 +383,9 @@ def __contains__(self, item):
 
     def __repr__(self):
         return ", ".join([str(i) for i in self.ranges])
+
+    def __eq__(self, other):
+        if len(self.ranges) != len(other.ranges):
+            return False
+
+        return all(i == j for i, j in zip(self.ranges, other.ranges))
```

Comment:
Just a note here: this assumes that the `Ranges` obtained contain `Range` objects in a monotonously increasing fashion. I'd think that that's okay.

```python
from strawberryfields.circuitspecs import Ranges

x = Ranges([0], [0.2, 0.55], [1.0])
y = Ranges([0], [1.0], [0.2, 0.55])

assert x == y
```
```
AssertionError                            Traceback (most recent call last)
<ipython-input-6-cdb229d2c821> in <module>
      4 y = Ranges([0], [1.0], [0.2, 0.55])
      5 
----> 6 assert x == y

AssertionError: 
```

Reply:
Good point @antal!